sw/cheri/tests/pinmux_tests.hh

```diff
@@ -0,0 +1,226 @@
+// Copyright lowRISC Contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+#include "../../common/defs.h"
+#include "../common/console.hh"
+#include "../common/flash-utils.hh"
+#include "../common/timer-utils.hh"
+#include "../common/uart-utils.hh"
+#include "../common/sonata-devices.hh"
+#include "../common/platform-pinmux.hh"
+#include "test_runner.hh"
+#include "i2c_tests.hh"
+#include <cheri.hh>
+#include <platform-uart.hh>
+#include <ds/xoroshiro.h>
+
+using namespace CHERI;
+
+/**
+ * Configures the number of test iterations to perform.
+ * This can be overriden via a compilation flag.
+ */
+#ifndef PINMUX_TEST_ITERATIONS
+#define PINMUX_TEST_ITERATIONS (1U)
+#endif
+
+/**
+ * Configures whether cable connections required for the pinmux testing
+ * are available. This includes cables between:
+ *  - mikroBus       (P7) RX & TX
+ * This can be overriden via a compilation flag.
+ */
+#ifndef PINMUX_CABLE_CONNECTIONS_AVAILABLE
+#define PINMUX_CABLE_CONNECTIONS_AVAILABLE true
+#endif
+
+static constexpr uint8_t PmxToDisabled         = 0;
+
+/**
+ * Attempt to retrieve the JEDEC ID of the Spi Flash device on the Sonata board,
+ * and compare it to the known (expected) JEDEC ID. This is separate from the
+ * similar test in the SPI tests, as it skips the initialisation logic that
+ * should not be repeated in the pinmux check.
+ *
+ * Returns the number of failures during the test.
+ */
+static int spi_jedec_id_test(Capability<volatile SonataSpi> spi, SpiFlash spi_flash) {
+  int failures = 0;
+
+  // Read the JEDEC ID from Flash
+  uint8_t jedec_id[3] = {0};
+  spi_flash.read_jedec_id(jedec_id);
+
+  // Check that the retrieved ID matches our expected value
+  for (size_t index = 0; index < 3; index++) {
+    if (jedec_id[index] != ExpectedSpiFlashJedecId[index]) {
+      failures++;
+    }
+  }
+
+  return failures;
+}
+
+/**
+ * Test pinmux by enabling and disabling the SPI Flash pins. First reads the
+ * Flash's JEDEC ID like normal, then disables the pins via pinmux and repeats
+ * the JEDEC ID read, checking that it fails. Then re-enables the pins via
+ * pinmux and repeats the JEDEC ID read, checking that it succeeds.
+ * Returns the number of failures during the test.
+ */
+static int pinmux_spi_flash_test(SonataPinmux *pinmux, Capability<volatile SonataSpi> spi, SpiFlash spi_flash) {
+  constexpr uint8_t PmxSpiFlashDataToSpiTx0   = 1;
+  constexpr uint8_t PmxSpiFlashClockToSpiClk0 = 1;
+
+  int failures = 0;
+
+  // Ensure the SPI Flash pins are enabled using Pinmux
+  if (!pinmux->output_pin_select(SonataPinmux::OutputPin::SpiFlashData, PmxSpiFlashDataToSpiTx0)) failures++;
```

Comment:
This is my personal preference, I don't like if statements without `{}`, so I would suggest:
failures += (condition == true);

Reply:
This makes sense, I'll convert these kinds of statements in a follow-up PR